rpcserver/socket_reads.cc

```diff
@@ -125,13 +127,26 @@ namespace rpcserver {
         delete[] buf;
         return !error && proto.ParseFromString(proto_str);
     }
-
-
+	
+	/**
+     * Attempt to parse given protocol with given length from provided socket.
+     * Return whether the parse was successful.
+     */
+    bool read_proto(tcp::socket& sock, ::google::protobuf::Message& proto, uint64_t len) {
+        asio::error_code error;
+        char* buf = new char[len];
```

Comment:
You can make a string with length len, and then pass string.c_str() into buffer. This way we can get rid of char * and delete.